### PR TITLE
[TTAHUB-1885] Update training and session report permissions

### PR DESCRIPTION
**Pull request by thewatermethod**
### Update permissions for the training report (based on current understanding)

**Event permissions**
- Event owner: can add sessions, edit event, edit and delete sessions, close or suspend event
- Event collaborator: can add sessions, edit and delete sessions. Cannot edit event.
- Regional point of contact: can edit event, edit and delete sessions. Cannot add sessions or close or suspend event.

**Event visibility**
- Not started: Only owners and those added to the event can view event and sessions
- In progress: Everyone with read or read/write can view event and sessions. Everyone else can see event but only completed sessions.
- Complete & Suspended: Utter visibility

### Chart for reference:

Screenshot 2023-07-25 at 8 46 06 AM

### Additional changes

This PR also includes a few small updates requested by the UX team.
- The topics control on the session summary now has a drawer. The language for that drawer, previously ambiguous, has been updated both here and on the AR.
- The instructions on the complete summary page have been updated
- Event creator has been added as a field to the "view event" page.

## How to test

Makes sure behavior conforms to expectations. Make sure behavior is adequately tested. Confirm smaller bugfixes.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1885
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1895
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1897
* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1900

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

**Comment by kryswisnaskas** [comment]
Thanks for fixing the error! I don't see the previous issue anymore. Things look good. 👍 

A couple of observations which can likely be addressed in different PRs

- When a collaborator edits a session we are giving them an option to move it from "Complete" to "In progress". Users are notified the session has been successfully saved, however the change of status doesn't take an effect anywhere. The session stays in the "Complete" state. Did we decide that collaborators can move a completed session to "In progress"?

- Auto save: Auto save doesn't seem to be implemented. We are notifying users that we autosaved when saving draft but we just save in this case.

**Comment by thewatermethod** [comment]
@kryswisnaskas 

> Thanks for fixing the error! I don't see the previous issue anymore. Things look good. 👍

I think you might be referring to the errors you found on #1636, a very similarly named PR. That PR renames the permission, this one changes all the permissions around to match the new understanding. #1636 does need another look, as I did address the issue you found.

>     * When a collaborator edits a session we are giving them an option to move it from "Complete" to "In progress". Users are notified the session has been successfully saved, however the change of status doesn't take an effect anywhere. The session stays in the "Complete" state. Did we decide that collaborators can move a completed session to "In progress"?

They can edit sessions, so yes. I just pushed a fix for that, thanks for finding that. 

>     * Auto save: Auto save doesn't seem to be implemented. We are notifying users that we autosaved when saving draft but we just save in this case.

Hmm, autosave is implemented. It inherits that functionality directly from the AR. Is it possible that your form wasn't "dirty"?

Ran a test below, saw a PUT request after ~2 minutes with an updated message in the sidebar.

Screenshot 2023-07-25 at 10 30 17 AM

File: frontend/src/pages/TrainingReports/components/EventCard.js [hunk 1]
```diff
@@ -31,23 +25,32 @@ function EventCard({
     sessionReports,
   } = event;
 
-  const contextMenuLabel = `Actions for event ${event.id}`;
-  const menuItems = [];
+  const isOwner = event.ownerId === user.id;
```
Comment (AdamAdHocTeam) [inline frontend/src/pages/TrainingReports/components/EventCard.js:28]: Once again

File: frontend/src/components/GoalForm/ObjectiveTopics.js [hunk 2]
```diff
@@ -96,7 +96,7 @@ export default function ObjectiveTopics({
               className="usa-button usa-button--unstyled margin-left-1"
               ref={drawerTriggerRef}
             >
-              Tell me more
+              Get help choosing topics
```
Comment (AdamAdHocTeam) [inline frontend/src/components/GoalForm/ObjectiveTopics.js:99]: Get some help

**Review by AdamAdHocTeam — APPROVED** [review]
Looks good.

**Pull request merged**